apollo-portal/src/main/java/com/ctrip/framework/apollo/openapi/v1/controller/NamespaceBranchController.java

```diff
@@ -0,0 +1,107 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+
+@RestController("openapiNamespaceBranchController")
+@RequestMapping("/openapi/v1/envs/{env}")
+public class NamespaceBranchController {
+
+    @Autowired
+    private NamespaceBranchService namespaceBranchService;
+
+    @RequestMapping(value = "/apps/{appId}/clusters/{clusterName}/namespaces/{namespaceName}/branches", method = RequestMethod.GET)
+    public OpenNamespaceDTO findBranch(@PathVariable String appId,
+                                       @PathVariable String env,
+                                       @PathVariable String clusterName,
+                                       @PathVariable String namespaceName) {
+        NamespaceBO namespaceBO = namespaceBranchService.findBranch(appId, Env.valueOf(env.toUpperCase()), clusterName, namespaceName);
+        if (namespaceBO == null) {
+            return null;
+        }
+        return OpenApiBeanUtils.transformFromNamespaceBO(namespaceBO);
+    }
+
+    @PreAuthorize(value = "@consumerPermissionValidator.hasCreateNamespacePermission(#request, #appId)")
+    @RequestMapping(value = "/apps/{appId}/clusters/{clusterName}/namespaces/{namespaceName}/branches", method = RequestMethod.POST)
+    public OpenNamespaceDTO createBranch(@PathVariable String appId,
+                                         @PathVariable String env,
+                                         @PathVariable String clusterName,
+                                         @PathVariable String namespaceName,
+                                         @RequestParam("operator") String operator,
+                                         HttpServletRequest request) {
+        NamespaceDTO namespaceDTO = namespaceBranchService.createBranch(appId, Env.valueOf(env.toUpperCase()), clusterName, namespaceName, operator);
+        if (namespaceDTO == null) {
+            return null;
+        }
+        return BeanUtils.transfrom(OpenNamespaceDTO.class, namespaceDTO);
+    }
+
+    @PreAuthorize(value = "@consumerPermissionValidator.hasModifyNamespacePermission(#request, #appId, #namespaceName, #env)")
+    @RequestMapping(value = "/apps/{appId}/clusters/{clusterName}/namespaces/{namespaceName}/branches/{branchName}", method = RequestMethod.DELETE)
+    public void deleteBranch(@PathVariable String appId,
+                             @PathVariable String env,
+                             @PathVariable String clusterName,
+                             @PathVariable String namespaceName,
+                             @PathVariable String branchName,
+                             @RequestParam("operator") String operator,
+                             HttpServletRequest request) {
+/**
+ TBD
+ boolean canDelete = permissionValidator.hasReleaseNamespacePermission(appId, namespaceName) ||
```

Comment:
may keep the original permission logic:

1. check release permission
2. check modification permission and whether the branch has been released